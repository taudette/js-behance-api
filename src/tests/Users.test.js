import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import UserContainer from '../containers/User'
import UsersContainer from '../containers/Users'
import UsersComponent from '../components/Users'

configure({ adapter: new Adapter() });

describe('Users container', () => {
  it('UsersContainer is connected', () => {
    const mockStore = configureStore([]);
    const store = mockStore({});
    const wrapper = shallow(<Provider store={store}>
      <UsersContainer />
    </Provider>);
    expect(wrapper.text()).toBe('<Connect(UsersContainer) />');
  })
})

describe('Users component', () => {
	const users = shallow(<UsersComponent />);
	it('Users component renders users list correctly', () => {
		expect(users).toMatchSnapshot();
	});
})
