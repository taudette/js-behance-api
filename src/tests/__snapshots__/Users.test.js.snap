// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component Users component renders users list correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "panel",
      "bsStyle": "default",
      "children": <Row
        bsClass="row"
        componentClass="div"
>
        <p>
                Sorry try a different search!
        </p>
</Row>,
      "defaultExpanded": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": Array [
          undefined,
          <p>
            Sorry try a different search!
</p>,
        ],
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry try a different search!",
          },
          "ref": null,
          "rendered": "Sorry try a different search!",
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "panel",
        "bsStyle": "default",
        "children": <Row
          bsClass="row"
          componentClass="div"
>
          <p>
                    Sorry try a different search!
          </p>
</Row>,
        "defaultExpanded": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": Array [
            undefined,
            <p>
              Sorry try a different search!
</p>,
          ],
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sorry try a different search!",
            },
            "ref": null,
            "rendered": "Sorry try a different search!",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
